pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -68,6 +68,37 @@ const (
 )
 
 const (
+	activateRawOsdOnNodeScript = `
```

Comment:
Why not append the logic to `activateOSDCode` instead of a new block? The file is already complex enough so if you could keep the same logic within the existing place it's a win :).
Also, the current implementation doesn't address the metadata device aspect.

Reply:
When I create OSDs with metadata on nodes, Rook uses LVM. I don't think it's necessary. But point taken.

Reply:
https://github.com/rook/rook/blob/ea97c62954853616169d7510011868eafbdebcf0/pkg/daemon/ceph/osd/volume.go#L537-L541

OSDs on PVC no longer use the activate script to activate OSDs. And raw mode OSDs cannot be created with metadata partitions. So this is not a case we have to consider.